### PR TITLE
Upgrade next RHPAM product version to 7.13.4

### PR DESCRIPTION
**Pull request by lampajr**
**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[link](https://examle.com)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* referenced pull request 1
* referenced pull request 2
* referenced pull request 2
etc. 

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

File: job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/Constants.groovy [hunk 1]
```diff
@@ -41,7 +41,7 @@ class Constants {
     static final String MAIN_BRANCH_PROD_VERSION = '7.14.0'
     static final String KOGITO_MAIN_BRANCH_PROD_VERSION = '2.0.0'
     static final String RHBOP_MAIN_BRANCH_PROD_VERSION = '8.30.0'
-    static final String NEXT_PROD_VERSION = '7.13.3'
+    static final String NEXT_PROD_VERSION = '7.13.4'
     static final String CURRENT_PROD_VERSION = '7.12.2'
```
Comment (mareknovotny) [inline job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/Constants.groovy:45]: This is also confusing value, is it for the minimal version for upgrade?

Comment (lampajr) [reply job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/Constants.groovy:45]: I would say they are old properties, I checked in the repo and I did not find any reference to it (neither to `BAMOE_CURRENT_PROD_VERSION`)

Comment (rgdoliveira) [reply job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/Constants.groovy:45]: @lampajr your PR Looks good related to the next prod version of RHPAM.

@davidesalerno @spolti related to `CURRENT_PROD_VERSION` and `BAMOE_CURRENT_PROD_VERSION` properties, are they used somewhere in your scripts? I also could not find any reference to those variables in the repository.

Comment (rgdoliveira) [reply job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/Constants.groovy:45]: additional info:
`BAMOE_CURRENT_PROD_VERSION` was added in https://github.com/kiegroup/kie-ci/pull/1425/files#diff-9d55ce9075a526a3684617ae54e1f15ee81673ba5c89794b916086a8ffacc4d9R47

`CURRENT_PROD_VERSION` already existed and had old value 7.12.0 and was updated to `7.12.21 in https://github.com/kiegroup/kie-ci/pull/1395/files#diff-9d55ce9075a526a3684617ae54e1f15ee81673ba5c89794b916086a8ffacc4d9R45

If they are really not used I guess we can remove both.

Comment (mareknovotny) [reply job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/Constants.groovy:45]: @spolti @davidesalerno could you please confirm you don't use that `static final String CURRENT_PROD_VERSION = '7.12.2'` anymore?

Comment (spolti) [reply job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/Constants.groovy:45]: IIRC this env is not used in the nightly images pipeline. We do use the NEXT_PROD_VERSION and BAMOE_NEXT_PROD_VERSION.

Comment (rgdoliveira) [reply job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/Constants.groovy:45]: @spolti so we can remove both BAMOE_CURRENT_PROD_VERSION and CURRENT_PROD_VERSION variables?

Comment (spolti) [reply job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/Constants.groovy:45]: I think so.

**Review by rgdoliveira — APPROVED** [review]

**Comment by rgdoliveira** [comment]
@mareknovotny @lampajr can we merge this one?

**Pull request merged**